src/db/migrations/20250225111207_add_feature_flag_events.js

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+export async function up(knex) {
+  await knex.schema.createTable("feature_flag_events", (table) => {
+    table.string("name").notNullable();
+    table.boolean("is_enabled").notNullable().defaultTo(false);
+    table.specificType("allow_list", "character varying(255)[]").nullable();
+    table.timestamp("created_at").notNullable().defaultTo(knex.fn.now());
+    table.integer("created_by_subscriber_id").notNullable();
+    table.index(["name", "created_at"]);
+
+    table
+      .foreign("created_by_subscriber_id")
+      .references("id")
+      .inTable("subscribers");
+  });
+
+  await knex.schema.createView("feature_flag_view", (view) => {
+    view.columns([
+      "name",
+      "is_enabled",
+      "allow_list",
+      "updated_at",
+      "last_updated_by_subscriber_id",
+    ]);
+    view.as(
+      knex("feature_flag_events")
+        .select(
+          "name",
+          "is_enabled",
+          "allow_list",
+          "created_at as updated_at",
+          "created_by_subscriber_id as last_updated_by_subscriber_id",
+        )
+        .distinctOn("name")
+        .orderBy([
+          { column: "name", order: "asc" },
+          { column: "created_at", order: "desc" },
+        ]),
+    );
+  });
+
+  const existingFeatureFlags = await knex("feature_flags").select(
+    "name",
+    "is_enabled",
+    "allow_list",
+    "modified_at as created_at",
+  );
+  const fakeFirstuser = await knex("subscribers")
+    .select("id")
+    .whereLike("primary_email", "%@mozilla.com")
+    .first();
+  if (!fakeFirstuser) {
+    throw new Error(
+      "Couldn't find a subscriber with an @mozilla.com email address to attribute pre-existing feature flags to.",
+    );
+  }
+  await knex("feature_flag_events").insert(
+    existingFeatureFlags.map((flag) => ({
+      ...flag,
+      created_by_subscriber_id: fakeFirstuser.id,
```

Comment:
Good to call out: we're currently not tracking who last touched a feature flag, so I just pretended that the first subscriber with an `@mozilla` email address did (this is probably Luke I suppose). Since this is for our eyes only, it doesn't really matter, I don't think.

Reply:
Oh, I guess this doesn't work on an empty DB, like in our e2e tests. I'd really prefer not to make the column nullable though...